devito/types/parallel.py

```diff
@@ -254,6 +254,14 @@ class DeviceID(DeviceSymbol):
     def default_value(self):
         return -1
 
+    def _arg_values(self, **kwargs):
+        if self.name in kwargs:
+            return {self.name: kwargs.pop(self.name)}
+        elif configuration['deviceid'] != self.default_value:
+            return {self.name: configuration['deviceid']}
+        else:
+            return {self.name: self.default_value}
```

Comment:
why not just `{self.name: kwargs.pop(self.name, self.default_value)}` and have `default_value` return `configuration['deviceid']`

Reply:
isn't that what the original code did? I thought the idea was to keep `default_value` and `configuration['deviceid']` separate?

Reply:
yes, agree with Carlos here. We can't just return a (mutable) global value from `default_value` :-)